brayns/engine/Scene.cpp

```diff
@@ -21,58 +21,41 @@
 
 #include "Scene.h"
 
-#include <brayns/engine/common/DataHandler.h>
+#include <ospray/ospray_cpp/Data.h>
 
 namespace
 {
-class InstanceCommitter
+class WorldParameterUpdater
 {
 public:
-    static void commit(OSPWorld world, const std::vector<OSPInstance> &instances)
+    static void updateInstances(const ospray::cpp::World &world, const std::vector<ospray::cpp::Instance> &instances)
     {
+        static const std::string instanceParameter = "instance";
+
         if (instances.empty())
         {
-            ospRemoveParam(world, "instance");
+            world.removeParam(instanceParameter);
             return;
         }
-
-        auto instanceBuffer = brayns::DataHandler::copyBuffer(instances, OSPDataType::OSP_INSTANCE);
-        ospSetParam(world, "instance", OSPDataType::OSP_DATA, &instanceBuffer.handle);
+        world.setParam(instanceParameter, ospray::cpp::CopiedData(instances));
     }
-};
 
-class LightComitter
-{
-public:
-    static void commit(OSPWorld world, const std::vector<OSPLight> &lights)
+    static void updateLights(const ospray::cpp::World &world, const std::vector<ospray::cpp::Light> &lights)
     {
+        static const std::string lightParameter = "light";
+
         if (lights.empty())
         {
-            ospRemoveParam(world, "light");
+            world.removeParam(lightParameter);
             return;
         }
-
-        auto lightBuffer = brayns::DataHandler::copyBuffer(lights, OSPDataType::OSP_LIGHT);
-        ospSetParam(world, "light", OSPDataType::OSP_DATA, &lightBuffer.handle);
+        world.setParam(lightParameter, ospray::cpp::CopiedData(lights));
```

Comment:
Just curiosity, why do we need to copy the lights to the world ? We cannot transfer ownership of these lights to te world (~move) ?

Reply:
We are not copying the lights, we are copying a list of pointers to the lights